src/parse_policies.c

```diff
@@ -138,3 +140,84 @@ int ParseCompactionPolicy(const char *policy_string,
     }
     return success;
 }
+
+// Helper function to convert milliseconds to a time string
+static inline void MillisecondsToTimeString(uint64_t millis, char *out, const size_t outLength) {
+    if (millis % (1000 * 60 * 60 * 24) == 0) {
+        snprintf(out, outLength, "%" PRIu64 "d", millis / (1000 * 60 * 60 * 24));
+    } else if (millis % (1000 * 60 * 60) == 0) {
+        snprintf(out, outLength, "%" PRIu64 "h", millis / (1000 * 60 * 60));
+    } else if (millis % (1000 * 60) == 0) {
+        snprintf(out, outLength, "%" PRIu64 "M", millis / (1000 * 60));
+    } else if (millis % 1000 == 0) {
+        snprintf(out, outLength, "%" PRIu64 "s", millis / 1000);
+    } else {
+        snprintf(out, outLength, "%" PRIu64 "m", millis);
```

Comment:
shouldn't minutes be `"m"` and milliseconds be `"ms"`? afaik, `"M"` is for months

Reply:
Yes, but the code isn't mine and changing it will break the compatibility. Someone who initially came up with this should have thought earlier.